src/app/youtube/services/sortbar-manager.service.ts

```diff
@@ -0,0 +1,73 @@
+import { Injectable } from '@angular/core'
+import { BehaviorSubject, Observable } from 'rxjs'
+import { ISortOptions, SortOrder, SortType } from '../models/sort-options.model'
+
+export interface ISortButtonsClasses {
+  activeDate: boolean
+  activeViewsCount: boolean
+  activeByKeywords: boolean
+  order: SortOrder
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class SortbarManagerService {
+  isSortVisible = false
```

Comment:
do you need this field if you already have visibility state stored in subject

Reply:
![image](https://user-images.githubusercontent.com/51916577/130364123-93905086-4aaf-47f7-ac81-c52842fa0aee.png)
you don't need isSortVisible field, when you already have sortVisibility$$ subject. I see that isSortVisible is used only once for toggle, you can do it using sortVisibility$$.value.